docs/rules/rule_md049.md

```diff
@@ -0,0 +1,21 @@
+# Rule - MD049
+
+| Aliases        |
+|----------------|
+| `md049`        |
+| validate-refs  |
+
+## Summary
+
+Ensure that all references to parts of the documentation are valid.
+
+## Reasoning
```

Comment:
for me, what is missing are some good positive examples and negative examples on this page to help people understand what this plugin does and does not do.

for example:
- does this plugin assume all assets in same directory?  if not, what rules are followed to get there? can you "escape" the project level?
- I noticed a test where you had "/" for one of the links.  can you call out that behavior? 
- only for markdown files is the "#" operator adhered to. Seeing as the identified can be used with other types of URL (html pages for one), can you add something saying the plugin only respects the # operator for files with a Markdown extension.